core/iwasm/compilation/aot_emit_aot_file.c

```diff
@@ -1645,9 +1651,11 @@ get_name_section_size(AOTCompData *comp_data)
                         previous_func_index = func_index;
                         read_leb_uint32(p, p_end, func_name_len);
                         offset = align_uint(offset, 2);
-                        EMIT_U16(func_name_len);
+                        /* extra 1 bytes for \0 */
+                        EMIT_U16(func_name_len + 1);
```

Comment:
Had better check whether `func_name_len < UINT16_MAX`, and use `EMIT_U16((uint16)func_name_len + 1)`? Here uint32 is truncated into uint16.

Reply:
`EMIT_U16()` will truncate the value.

Not sure about the check. Since we are behind loader, and it is an valid .wasm. If `func_name_len >= UINT16_MAX`, it will be a to big wasm file to load.